pkg/operator/controllers/muo/deploy_test.go

```diff
@@ -0,0 +1,126 @@
+func TestDeployCreateOrUpdate(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	setPullSpec := "MyMUOPullSpec"
+	cluster := &arov1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: arov1alpha1.SingletonClusterName,
+		},
+		Spec: arov1alpha1.ClusterSpec{
+			OperatorFlags: arov1alpha1.OperatorFlags{
+				pullSpec: setPullSpec,
+			},
+		},
+	}
+
+	k8scli := fake.NewSimpleClientset()
+	dh := mock_dynamichelper.NewMockInterface(controller)
+
+	// When the DynamicHelper is called, store the object kinds it creates.
+	var deployments []*appsv1.Deployment
+	deployedObjects := make(map[string]bool)
+	accessor := meta.NewAccessor()
+	check := func(ctx context.Context, objs ...kruntime.Object) error {
+		for _, i := range objs {
+			kind, err := accessor.Kind(i)
+			if err != nil {
+				t.Error(err)
+			}
+			if d, ok := i.(*appsv1.Deployment); ok {
+				deployments = append(deployments, d)
+			}
+			deployedObjects[kind] = true
+		}
+		return nil
+	}
+	dh.EXPECT().Ensure(gomock.Any(), gomock.Any()).Do(check).Return(nil)
+
+	deployer := NewMUODeployer(cluster, k8scli, dh)
+	err := deployer.CreateOrUpdate(context.Background())
+	if err != nil {
+		t.Error(err)
+	}
+
+	expectedKinds := map[string]bool{
+		"ClusterRole":              true,
+		"ConfigMap":                true,
+		"ClusterRoleBinding":       true,
+		"CustomResourceDefinition": true,
+		"Deployment":               true,
+		"Namespace":                true,
+		"Role":                     true,
+		"RoleBinding":              true,
+		"ServiceAccount":           true,
+	}
+	errs := deep.Equal(deployedObjects, expectedKinds)
+	for _, e := range errs {
+		t.Error(e)
+	}
+
+	// Ensure we have set the pullspec set on the containers
+	for _, d := range deployments {
+		for _, c := range d.Spec.Template.Spec.Containers {
+			if c.Image != setPullSpec {
+				t.Errorf("expected %s, got %s for pullspec", setPullSpec, c.Image)
+			}
+		}
+	}
+}
+
+func TestDeployDelete(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	cluster := &arov1alpha1.Cluster{}
+	k8scli := fake.NewSimpleClientset()
+
+	dh := mock_dynamichelper.NewMockInterface(controller)
+
+	// keep track of deleted objects
+	deleted := []string{}
+	checkDelete := func(ctx context.Context, kind string, namespace string, name string) error {
+		deleted = append(deleted, fmt.Sprintf("%s/%s/%s", kind, namespace, name))
+		return nil
+	}
+	dh.EXPECT().EnsureDeleted(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes().Do(checkDelete).Return(nil)
+
+	deployer := NewMUODeployer(cluster, k8scli, dh)
+	errs := deployer.Remove(context.Background())
+	if len(errs) != 0 {
+		t.Error(errs)
+	}
+
+	expectedDeleted := []string{
+		"Deployment/openshift-managed-upgrade-operator/managed-upgrade-operator",
+	}
+	sort.Strings(deleted)
+	sort.Strings(expectedDeleted)
+	mismatches := deep.Equal(deleted, expectedDeleted)
+	for _, e := range mismatches {
+		t.Error(e)
+	}
```

Comment:
Go mock is great in simple cases like this ;)

```suggestion
	controller := gomock.NewController(t)
	defer controller.Finish()

	cluster := &arov1alpha1.Cluster{}
	k8scli := fake.NewSimpleClientset()

	dh := mock_dynamichelper.NewMockInterface(controller)
	dh.EXPECT().EnsureDeleted(gomock.Any(), "Deployment", "openshift-managed-upgrade-operator", "managed-upgrade-operator").Return(nil)

	deployer := NewMUODeployer(cluster, k8scli, dh)
	errs := deployer.Remove(context.Background())
	if len(errs) != 0 {
		t.Error(errs)
	}
```